src/Func.h

```diff
@@ -216,13 +216,60 @@ class Stage {
     EXPORT Stage &gpu(VarOrRVar block_x, VarOrRVar block_y, VarOrRVar block_z,
                       VarOrRVar thread_x, VarOrRVar thread_y, VarOrRVar thread_z,
                       DeviceAPI device_api = DeviceAPI::Default_GPU);
+
+    // TODO(psuriana): For now we need to expand "tx" into Var and RVar versions
```

Comment:
Never mind, I see now that the problem was with implicitly converting Var to Expr, not to a different VarOrRVar overload.